src/tags/for.ts

```diff
@@ -78,6 +79,30 @@ export default class extends Tag {
     }
     ctx.pop()
   }
+
+  public node (): StaticNode {
+    const children = this.templates
+
+    if (this.elseTemplates) {
+      children.push(...this.elseTemplates)
+    }
+
+    const values: Array<Value | ValueToken> = [this.collection]
+
+    for (const v of Object.values(this.hash.hash)) {
+      if (isValueToken(v)) {
+        values.push(v)
+      }
+    }
+
+    return {
+      token: this.token,
+      values,
```

Comment:
question: is it possible to get `values` by recursively iterate `.token` and `.children`? if so, can we remove exposed `values` on each tag and only retrieve it during `analyze`?

Reply:
We could make `analyze()` "aware" of the standard tags, and have it "know" what properties might contain values (those properties vary quite a bit). But that would not generalize to all possible custom tags (including custom versions of standard tags), so we'd still need a way for unknown tags to tell us what values they manage.

And if we took a hybrid approach, where our recursive analysis routine reads some properties directly and relies on `Node.values` for others, whenever someone wants to traverse the template for their own purposes, they too would need to include that knowledge of the built in tags.

Reply:
Agree with hybrid approach would create more confusion. By `values they manage`, I guess you mean like arguments for a tag. It's now technically being children so we may want them be exposed by another property other than `children`. I'm not sure whether it's a good idea to expose another prop named `arguments`, is this sufficient? As in my understanding, a template consists of name, args and children.

The idea behind it is, it'll be more useful if `node` intend to expose an AST structure, which can be used more generally apart from analyzing variables.

Reply:
Yeah, `values` is essentially the tag's arguments. There might be one or two cases where an argument is not a value (as in an instance of the `Value` class or `ValueToken` type), but the distinction is not really important. We could rename `values` to `arguments`.

I hadn't intended nodes returned from `Template.node()` (I should rename that) to form an AST. In my head it is more of an _auxiliary tree_, specifically designed to carry template information useful for static analysis in the absence of an AST and without introducing breaking changes.